src/components/molecules/ExecutionDetails/TestExecutionDetails/TestExecutionDetailsTabs.tsx

```diff
@@ -83,39 +91,64 @@ const TestExecutionDetailsTabs: React.FC = () => {
     }
   }, [isRunning, id]);
 
-  const items = [
+  const defaultExecutionDetailsTabs = [
     {
-      key: 'LogOutputPane',
-      label: 'Log Output',
-      children: <LogOutput logOutput={output} executionId={id} isRunning={isRunning} isAutoScrolled={isAutoScrolled} />,
+      value: {
+        key: 'LogOutputPane',
+        label: 'Log Output',
+        children: (
+          <LogOutput logOutput={output} executionId={id} isRunning={isRunning} isAutoScrolled={isAutoScrolled} />
+        ),
+      },
+      metadata: {
+        order: Infinity,
+      },
     },
+    // TODO: refactor using visibility metadata
     whetherToShowArtifactsTab
       ? {
-          key: 'ArtifactsPane',
-          label: 'Artifacts',
-          children: (
-            <TestExecutionDetailsArtifacts
-              id={id}
-              testName={testName}
-              testSuiteName={testSuiteName}
-              startTime={startTime.toString()}
-            />
-          ),
+          value: {
+            key: 'ArtifactsPane',
+            label: 'Artifacts',
+            children: (
+              <TestExecutionDetailsArtifacts
+                id={id}
+                testName={testName}
+                testSuiteName={testSuiteName}
+                startTime={startTime.toString()}
+              />
+            ),
+          },
+          metadata: {
+            order: 3,
+          },
         }
       : null,
     {
-      key: 'CLICommands',
-      label: 'CLI Commands',
-      children: <CLICommands isExecutions type={testType} id={id} modifyMap={{status}} />,
+      value: {
+        key: 'CLICommands',
+        label: 'CLI Commands',
+        children: <CLICommands isExecutions type={testType} id={id} modifyMap={{status}} />,
+      },
+      metadata: {
+        order: 2,
+      },
     },
     decomposedVars.length
       ? {
-          key: 'Variables',
-          label: 'Variables',
-          children: <ExecutionsVariablesList variables={decomposedVars} />,
+          value: {
```

Comment:
This whole array should be filtered out of nullish values (`.filter(Boolean)`), otherwise `orderArray` will fail